src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BankHolidayCacheSchedule.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.calculatereleasedatesapi.service
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+
+@Component
+class BankHolidayCacheSchedule(
+  val bankHolidayService: BankHolidayService
+) {
+
+  // Every 5 mins from 03:00 till 03:59 Monday to Friday.
+  @Scheduled(cron = "0 */5 3 * * MON-FRI")
```

Comment:
In a clustered infrastructure where we have multiple pods (i.e. kubernetes) this will result in this schedule firing multiple times simultaneously (per pod)

There are several ways to solve this problem. On Manage Offences we have used SchedLock (See MO for example)

Reply:
Yea, so the code was written so that if the scheduled job was picked up by two different services at the same time, the logic would still work and update the cache.

Reply:
If multiple pods run simultaneously then isn't there a chance that they will both insert - because the commit isn't done until the transaction is finished? To discuss

Reply:
Yea that could happen. But only on the first ever run of the schedule. Subsequent runs will be updates, and the updates will just update the first row and the second row would be ignored forever.

I had a quick look and maybe a simple thing to do would make the second the job first on random.

https://stackoverflow.com/a/41018664

  @Scheduled(cron = "${random.int[0,59]} */5 3 * * MON-FRI")

Reply:
sounds good to me